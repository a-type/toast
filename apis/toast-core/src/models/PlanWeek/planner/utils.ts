import Schedule, { ScheduleMeal } from 'models/Schedule';
import { PlanWeekData } from '../types';

const MEALS_PER_DAY = 3;
const DAYS_PER_WEEK = 7;

export const mealsFromGroceryDay = (schedule: Schedule): ScheduleMeal[] => {
  return [
    ...schedule.meals.slice(schedule.groceryDay * MEALS_PER_DAY),
    ...schedule.meals.slice(0, schedule.groceryDay * MEALS_PER_DAY),
  ];
};

const findCookedMeal = meal =>
  meal.mealIndex !== 0 &&
  ['LONG', 'MEDIUM', 'SHORT'].includes(meal.availability);

export const initializeWeek = (
  schedule: Schedule,
  weekIndex: number,
  findFirstMeal: (meal: ScheduleMeal) => boolean = findCookedMeal,
) => {
  // skip breakfast for main planning
  // find first cook day after grocery day to use to initialize the week
  const firstCookMeal = mealsFromGroceryDay(schedule).find(findFirstMeal);

  if (!firstCookMeal) {
    throw new Error(
      "Sorry, we couldn't figure out where to start this plan based on your schedule.",
    );
  }

  const week: PlanWeekData = {
    id: `${schedule.id}_w_${weekIndex}`,
    weekIndex,
    mealsPerDay: MEALS_PER_DAY,
    startDay: firstCookMeal.dayIndex,
    startMeal: firstCookMeal.mealIndex,
    meals: new Array(MEALS_PER_DAY * DAYS_PER_WEEK)
      .fill(null)
      .map((_, idx) => ({
        id: `${schedule.id}_w_${weekIndex}_m_${idx}`,
        weekIndex,
        // offsets meal index according to the meal the week starts on
        mealIndex: (firstCookMeal.mealIndex + idx) % MEALS_PER_DAY,
        dayIndex: Math.floor(
          ((firstCookMeal.dayIndex * MEALS_PER_DAY +
            firstCookMeal.mealIndex +
            idx) /
            MEALS_PER_DAY) %
            DAYS_PER_WEEK,
        ),
        actions: [],
      })),
  };

  return week;
};
