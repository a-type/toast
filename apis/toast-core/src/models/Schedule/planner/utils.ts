import { ScheduleMeal } from 'models/Schedule/Schedule';
import { Meal, Schedule } from 'models';

export const MEALS_PER_DAY = 3;
export const DAYS_PER_WEEK = 7;

export const initializeWeek = (): Meal[] => {
  const meals = new Array(DAYS_PER_WEEK)
    .fill(null)
    .map((_, dayIndex) =>
      new Array(MEALS_PER_DAY).fill(null).map((__, mealIndex) => {
        return Meal.createEmpty(dayIndex, mealIndex);
      }),
    )
    .reduce((allMeals, dayMeals) => allMeals.concat(dayMeals), []);

  return meals;
};
