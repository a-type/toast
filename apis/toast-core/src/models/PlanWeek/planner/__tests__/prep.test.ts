import plan from '../index';
import plans from './createPlans';
import { ignoreIds } from './utils';
const {
  variety,
  largeFamily,
  busy,
  nefarious,
  basic,
  midweekGrocery,
} = plans();

describe('prep meal plan', () => {
  test('for a variety plan', () => {
    variety.strategy = 'PREP';
    expect(ignoreIds(plan.run(variety, 0).toJSON())).toMatchSnapshot();
  });

  test('for midweek grocery day', () => {
    midweekGrocery.strategy = 'PREP';
    expect(ignoreIds(plan.run(midweekGrocery, 0).toJSON())).toMatchSnapshot();
  });

  test('for a large family plan', () => {
    largeFamily.strategy = 'PREP';
    expect(ignoreIds(plan.run(largeFamily, 0).toJSON())).toMatchSnapshot();
  });

  test('for a really busy person', () => {
    busy.strategy = 'PREP';
    expect(ignoreIds(plan.run(busy, 0).toJSON())).toMatchSnapshot();
  });

  test('a basic plan', () => {
    basic.strategy = 'PREP';
    expect(ignoreIds(plan.run(basic, 0).toJSON())).toMatchSnapshot();
  });

  describe('nefarious cases', () => {
    test('no time', () => {
      nefarious.noTime.strategy = 'PREP';
      expect(() =>
        ignoreIds(plan.run(nefarious.noTime, 0).toJSON()),
      ).toThrowErrorMatchingInlineSnapshot(
        `"Sorry, we couldn't figure out where to start this plan based on your schedule."`,
      );
    });

    test('eating out all the time', () => {
      nefarious.eatOut.strategy = 'PREP';
      expect(() =>
        ignoreIds(plan.run(nefarious.eatOut, 0).toJSON()),
      ).toThrowErrorMatchingInlineSnapshot(
        `"Sorry, we couldn't figure out where to start this plan based on your schedule."`,
      );
    });
  });
});
