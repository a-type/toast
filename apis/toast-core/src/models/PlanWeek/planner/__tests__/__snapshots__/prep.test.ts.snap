// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prep meal plan a basic plan 1`] = `
Object {
  "meals": Array [
    Object {
      "actions": Array [
        Object {
          "dayIndex": 0,
          "mealIndex": 1,
          "mealType": "NORMAL",
          "servings": 4,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 0,
          "leftovers": false,
          "mealIndex": 1,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 0,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 0,
          "mealIndex": 2,
          "mealType": "FANCY",
          "servings": 2,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 0,
          "leftovers": false,
          "mealIndex": 2,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 0,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 1,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 1,
          "leftovers": true,
          "mealIndex": 1,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 1,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 1,
          "mealIndex": 2,
          "mealType": "FANCY",
          "servings": 4,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 1,
          "leftovers": false,
          "mealIndex": 2,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 1,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 2,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 2,
          "leftovers": true,
          "mealIndex": 1,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 2,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 2,
          "mealIndex": 2,
          "mealType": "FANCY",
          "servings": 4,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 2,
          "leftovers": false,
          "mealIndex": 2,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 2,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 3,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 3,
          "leftovers": true,
          "mealIndex": 1,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 3,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 3,
          "mealIndex": 2,
          "mealType": "NORMAL",
          "servings": 6,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 3,
          "leftovers": false,
          "mealIndex": 2,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 3,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 4,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 4,
          "leftovers": true,
          "mealIndex": 1,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 4,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 4,
          "mealIndex": 2,
          "mealType": "QUICK",
          "servings": 2,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 4,
          "leftovers": false,
          "mealIndex": 2,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 4,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 5,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 5,
          "leftovers": true,
          "mealIndex": 1,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 5,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 5,
          "mealIndex": 2,
          "mealType": "QUICK",
          "servings": 2,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 5,
          "leftovers": false,
          "mealIndex": 2,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 5,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 6,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 6,
          "mealIndex": 1,
          "mealType": "NORMAL",
          "servings": 2,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 6,
          "leftovers": false,
          "mealIndex": 1,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 6,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 6,
          "mealIndex": 2,
          "mealType": "FANCY",
          "servings": 2,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 6,
          "leftovers": false,
          "mealIndex": 2,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 6,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 0,
      "mealIndex": 0,
    },
  ],
  "mealsPerDay": 3,
  "startDay": 0,
  "startMeal": 1,
  "weekIndex": 0,
}
`;

exports[`prep meal plan for a large family plan 1`] = `
Object {
  "meals": Array [
    Object {
      "actions": Array [
        Object {
          "dayIndex": 0,
          "mealIndex": 2,
          "mealType": "BIG",
          "servings": 12,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 0,
          "leftovers": false,
          "mealIndex": 2,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 0,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 1,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 1,
          "mealIndex": 1,
          "mealType": "QUICK",
          "servings": 6,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 1,
          "leftovers": false,
          "mealIndex": 1,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 1,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 1,
          "mealIndex": 2,
          "mealType": "FANCY",
          "servings": 6,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 1,
          "leftovers": false,
          "mealIndex": 2,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 1,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 2,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 2,
          "mealIndex": 1,
          "mealType": "QUICK",
          "servings": 6,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 2,
          "leftovers": false,
          "mealIndex": 1,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 2,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 2,
          "mealIndex": 2,
          "mealType": "QUICK",
          "servings": 6,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 2,
          "leftovers": false,
          "mealIndex": 2,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 2,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 3,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 3,
          "leftovers": true,
          "mealIndex": 1,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 3,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 3,
          "mealIndex": 2,
          "type": "EAT_OUT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 3,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 4,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 4,
          "mealIndex": 1,
          "mealType": "QUICK",
          "servings": 6,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 4,
          "leftovers": false,
          "mealIndex": 1,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 4,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 4,
          "mealIndex": 2,
          "mealType": "BIG",
          "servings": 12,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 4,
          "leftovers": false,
          "mealIndex": 2,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 4,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 5,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 5,
          "leftovers": true,
          "mealIndex": 1,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 5,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 5,
          "mealIndex": 2,
          "mealType": "BIG",
          "servings": 12,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 5,
          "leftovers": false,
          "mealIndex": 2,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 5,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 6,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 6,
          "leftovers": true,
          "mealIndex": 1,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 6,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 6,
          "mealIndex": 2,
          "mealType": "FANCY",
          "servings": 6,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 6,
          "leftovers": false,
          "mealIndex": 2,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 6,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 0,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 0,
          "mealIndex": 1,
          "type": "EAT_OUT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 0,
      "mealIndex": 1,
    },
  ],
  "mealsPerDay": 3,
  "startDay": 0,
  "startMeal": 2,
  "weekIndex": 0,
}
`;

exports[`prep meal plan for a really busy person 1`] = `
Object {
  "meals": Array [
    Object {
      "actions": Array [
        Object {
          "dayIndex": 0,
          "mealIndex": 1,
          "mealType": "QUICK",
          "servings": 2,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 0,
          "leftovers": false,
          "mealIndex": 1,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 0,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 0,
          "mealIndex": 2,
          "mealType": "QUICK",
          "servings": 2,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 0,
          "leftovers": false,
          "mealIndex": 2,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 0,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 1,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 1,
          "mealIndex": 1,
          "type": "EAT_OUT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 1,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [],
      "dayIndex": 1,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 2,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [],
      "dayIndex": 2,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 2,
          "mealIndex": 2,
          "mealType": "QUICK",
          "servings": 2,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 2,
          "leftovers": false,
          "mealIndex": 2,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 2,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 3,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [],
      "dayIndex": 3,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 3,
          "mealIndex": 2,
          "type": "EAT_OUT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 3,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 4,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [],
      "dayIndex": 4,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 4,
          "mealIndex": 2,
          "mealType": "QUICK",
          "servings": 8,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 4,
          "leftovers": false,
          "mealIndex": 2,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 4,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 5,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 5,
          "leftovers": true,
          "mealIndex": 1,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 5,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 5,
          "mealIndex": 2,
          "mealType": "QUICK",
          "servings": 2,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 5,
          "leftovers": false,
          "mealIndex": 2,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 5,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 6,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 6,
          "leftovers": true,
          "mealIndex": 1,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 6,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 6,
          "leftovers": true,
          "mealIndex": 2,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 6,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 0,
      "mealIndex": 0,
    },
  ],
  "mealsPerDay": 3,
  "startDay": 0,
  "startMeal": 1,
  "weekIndex": 0,
}
`;

exports[`prep meal plan for a variety plan 1`] = `
Object {
  "meals": Array [
    Object {
      "actions": Array [
        Object {
          "dayIndex": 0,
          "mealIndex": 1,
          "mealType": "QUICK",
          "servings": 2,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 0,
          "leftovers": false,
          "mealIndex": 1,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 0,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 0,
          "mealIndex": 2,
          "mealType": "FANCY",
          "servings": 4,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 0,
          "leftovers": false,
          "mealIndex": 2,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 0,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 1,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 1,
          "mealIndex": 1,
          "type": "EAT_OUT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 1,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 1,
          "mealIndex": 2,
          "mealType": "NORMAL",
          "servings": 4,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 1,
          "leftovers": false,
          "mealIndex": 2,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 1,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 2,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 2,
          "mealIndex": 1,
          "mealType": "QUICK",
          "servings": 2,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 2,
          "leftovers": false,
          "mealIndex": 1,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 2,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 2,
          "mealIndex": 2,
          "mealType": "QUICK",
          "servings": 2,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 2,
          "leftovers": false,
          "mealIndex": 2,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 2,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 3,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 3,
          "leftovers": true,
          "mealIndex": 1,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 3,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 3,
          "mealIndex": 2,
          "type": "EAT_OUT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 3,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 4,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 4,
          "leftovers": true,
          "mealIndex": 1,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 4,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 4,
          "mealIndex": 2,
          "mealType": "NORMAL",
          "servings": 4,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 4,
          "leftovers": false,
          "mealIndex": 2,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 4,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 5,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 5,
          "mealIndex": 1,
          "mealType": "QUICK",
          "servings": 2,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 5,
          "leftovers": false,
          "mealIndex": 1,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 5,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 5,
          "mealIndex": 2,
          "mealType": "FANCY",
          "servings": 2,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 5,
          "leftovers": false,
          "mealIndex": 2,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 5,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 6,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 6,
          "leftovers": true,
          "mealIndex": 1,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 6,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 6,
          "mealIndex": 2,
          "mealType": "FANCY",
          "servings": 2,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 6,
          "leftovers": false,
          "mealIndex": 2,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 6,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 0,
      "mealIndex": 0,
    },
  ],
  "mealsPerDay": 3,
  "startDay": 0,
  "startMeal": 1,
  "weekIndex": 0,
}
`;

exports[`prep meal plan for midweek grocery day 1`] = `
Object {
  "meals": Array [
    Object {
      "actions": Array [
        Object {
          "dayIndex": 1,
          "mealIndex": 2,
          "mealType": "NORMAL",
          "servings": 6,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 1,
          "leftovers": false,
          "mealIndex": 2,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 1,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 2,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 2,
          "mealIndex": 1,
          "mealType": "QUICK",
          "servings": 2,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 2,
          "leftovers": false,
          "mealIndex": 1,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 2,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 2,
          "mealIndex": 2,
          "mealType": "QUICK",
          "servings": 2,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 2,
          "leftovers": false,
          "mealIndex": 2,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 2,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 3,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 3,
          "leftovers": true,
          "mealIndex": 1,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 3,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 3,
          "mealIndex": 2,
          "type": "EAT_OUT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 3,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 4,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 4,
          "leftovers": true,
          "mealIndex": 1,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 4,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 4,
          "mealIndex": 2,
          "mealType": "NORMAL",
          "servings": 4,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 4,
          "leftovers": false,
          "mealIndex": 2,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 4,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 5,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 5,
          "mealIndex": 1,
          "mealType": "QUICK",
          "servings": 2,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 5,
          "leftovers": false,
          "mealIndex": 1,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 5,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 5,
          "mealIndex": 2,
          "mealType": "FANCY",
          "servings": 2,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 5,
          "leftovers": false,
          "mealIndex": 2,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 5,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 6,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 6,
          "leftovers": true,
          "mealIndex": 1,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 6,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 6,
          "mealIndex": 2,
          "mealType": "FANCY",
          "servings": 2,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 6,
          "leftovers": false,
          "mealIndex": 2,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 6,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 0,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 0,
          "mealIndex": 1,
          "mealType": "QUICK",
          "servings": 2,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 0,
          "leftovers": false,
          "mealIndex": 1,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 0,
      "mealIndex": 1,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 0,
          "mealIndex": 2,
          "mealType": "FANCY",
          "servings": 2,
          "type": "COOK",
          "weekIndex": 0,
        },
        Object {
          "dayIndex": 0,
          "leftovers": false,
          "mealIndex": 2,
          "type": "EAT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 0,
      "mealIndex": 2,
    },
    Object {
      "actions": Array [],
      "dayIndex": 1,
      "mealIndex": 0,
    },
    Object {
      "actions": Array [
        Object {
          "dayIndex": 1,
          "mealIndex": 1,
          "type": "EAT_OUT",
          "weekIndex": 0,
        },
      ],
      "dayIndex": 1,
      "mealIndex": 1,
    },
  ],
  "mealsPerDay": 3,
  "startDay": 4,
  "startMeal": 14,
  "weekIndex": 0,
}
`;
